### PR TITLE
Cleanups

### PR DESCRIPTION
**Pull request by felipec**
While doing some tests I noticed many areas of improvement.

These patches cleanup and simply the code.

There should be no functional changes.

### CODE REVIEW

File: hg-fast-export.py [hunk 1]
```diff
@@ -375,19 +376,19 @@ def get_branchname(name):
...
 
 def export_note(ui,repo,revision,count,authors,encoding,is_first):
-  (revnode,_,user,(time,timezone),_,_,_,_)=get_changeset(ui,repo,revision,authors,encoding)
-  if repo[revnode].hidden():
-    return count
+  ctx = repo[revision]
 
-  parents = [p for p in repo.changelog.parentrevs(revision) if p >= 0]
+  (_,user,(time,timezone),_,_,_,_)=get_changeset(ui,repo,revision,authors,encoding)
+  if ctx.hidden():
```
Comment (felipec) [inline hg-fast-export.py:382]: I just noticed this `ctx.hidden()` can also be done before `get_changeset()` now.

Comment (frej) [reply hg-fast-export.py:382]: Feel free to ammend this commit or add a new one.

**Review by frej — COMMENTED** [review]
LGTM, but the commit message for `Share the changectx more` says: "It's used everyone, might as well pass it along.", shouldn't it be "It's used everywhere" or "It's used many times, so do some manual CSE".

**Comment by felipec** [comment]
OK. I've updated the branch.

**Pull request closed without merging**